src/crimson/common/config_proxy.cc

```diff
@@ -23,6 +23,25 @@ ConfigProxy::ConfigProxy(const EntityName& name, std::string_view cluster)
 				  "keyring", "$osd_data/keyring");
   }
 }
+ConfigProxy::ConfigProxy(const EntityName& name, std::string_view cluster, const ConfigValues& config_values)
```

Comment:
instead of replicating the whole ctor of `ConfigProxy::ConfigProxy()`. i'd suggest 

1. modify the existing constructor so it accepts an (optional) `unique_ptr<ConfigValues>`
1. and reusing it using [delegating constructor](https://en.cppreference.com/w/cpp/language/initializer_list#Delegating_constructor) .